flink-core/src/main/java/org/apache/flink/configuration/ConfigOptions.java

```diff
@@ -270,7 +283,33 @@ public final ConfigOption<List<E>> defaultValues(E... values) {
          * @return The config option without a default value.
          */
         public ConfigOption<List<E>> noDefaultValue() {
-            return new ConfigOption<>(key, clazz, ConfigOption.EMPTY_DESCRIPTION, null, true);
+            return new ConfigOption<>(key, clazz, ConfigOption.EMPTY_DESCRIPTION, null, LIST);
+        }
+    }
+
+    /** Builder for map type {@link ConfigOption} with a value type V. */
+    public static class MapConfigOptionBuilder<V> {
+        private final String key;
+        private final Class<V> clazz;
+
+        MapConfigOptionBuilder(String key, Class<V> clazz) {
+            this.key = key;
+            this.clazz = clazz;
+        }
+
+        /** Defines that the option's type should be a list of previously defined atomic type. */
+        @SuppressWarnings("rawtypes")
```

Comment:
is there a particular reason you don't cast to `ListConfigOptionBuilder<Map<String, V>>`?

Reply:
The compiler doesn't let me.